Source/CarthageKit/GitHub.swift

```diff
@@ -265,6 +268,28 @@ private func fetchAllPages(URL: NSURL, credentials: GitHubCredentials?) -> ColdS
 			let thisData = ColdSignal.single(data)
 
 			if let HTTPResponse = response as? NSHTTPURLResponse {
+				if HTTPResponse.statusCode != 200 && HTTPResponse.statusCode != 404 {
+					return thisData
+						.tryMap { data, error -> AnyObject? in
+							return NSJSONSerialization.JSONObjectWithData(data, options: nil, error: error)
+						}
+						.map { dictionary -> String in
+							let message = JSONValue.parse(dictionary)["message"]
```

Comment:
Can we create a real model structure for this? It would clean up this parsing code, clarify intention, and might come in handy if we want to use fields other than `message` in the future.